### PR TITLE
Update artwork typography

### PR DESCRIPTION
**Pull request by zephraph**
This is _quite_ a significant change. Lots of text updates and lots of places that could possibly go wrong.

Most of this was straight conversion. I _did_ alter how h1's are rendered from the artist's name. Previously we had it where if there were multiple artists then there would be multiple h1's which is invalid from an SEO perspective.

There are also some accessibility lint rules that were failing around links that don't have hrefs but simply act as buttons. I disabled them for now, but I'd be open to fixing in this PR or a follow up. @dzucconi your thoughts would be welcome here.

Before:

image

After:

image

Before:

image

After:

image

Before:

image

After:

image

### CODE REVIEW

File: src/v2/Apps/Artwork/Components/ArtworkImageBrowser/ArtworkSharePanel.tsx [hunk 1]
```diff
@@ -128,9 +128,11 @@ export class ArtworkSharePanel extends React.Component<
               onClick={this.handleCopy}
             />
           </SansGrow>
-          <Sans size={["3", "2"]} weight="medium" color="black60">
+          <Text variant="mediumText" color="black60">
+            {/* FIXME Remove lint ignore */}
+            {/* eslint-disable-next-line  */}
             <a onClick={this.handleCopy}>{this.state.copyLabelText}</a>
```
Comment (dzucconi) [inline src/v2/Apps/Artwork/Components/ArtworkImageBrowser/ArtworkSharePanel.tsx:134]: You *should* be able to just swap these with `Clickable` and it work out just fine.

Comment (zephraph) [reply src/v2/Apps/Artwork/Components/ArtworkImageBrowser/ArtworkSharePanel.tsx:134]: Does the onClick move to the clickable?

File: src/v2/Apps/Artwork/Components/ArtworkSidebar/ArtworkSidebarExtraLinks.tsx [hunk 2]
```diff
@@ -1,4 +1,6 @@
-import { Box, Link, Sans, Separator, Spacer } from "@artsy/palette"
+// FIXME -- Remove this lint disable
+/* eslint-disable jsx-a11y/anchor-is-valid */
```
Comment (dzucconi) [inline src/v2/Apps/Artwork/Components/ArtworkSidebar/ArtworkSidebarExtraLinks.tsx:2]: Yeah this one looks more complicated and can/should be addressed in a separate PR. For actual anchor tags that *also* have `onClick` handlers you want to: tabIndex={0} to make them focusable, then add a keypress handler to trigger the same onClick action with enter/spacebar.

File: src/v2/Apps/Artwork/Components/TrustSignals/TrustSignal.tsx [hunk 3]
```diff
@@ -19,14 +19,16 @@ export const TrustSignal: FC<TrustSignalProps> = ({
     <Flex {...other}>
       {Icon}
       <Flex flexDirection="column" ml={1}>
+        {/* FIXME: Remove this lint ignore */}
+        {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
         <Link onClick={onClick}>
```
Comment (dzucconi) [inline src/v2/Apps/Artwork/Components/TrustSignals/TrustSignal.tsx:24]: Hm, I wonder if we could just `Link as="button"` with these actually; since they don't actually have hrefs.

File: src/v2/Apps/Artwork/Components/ArtworkImageBrowser/ArtworkSharePanel.tsx (hunk 1 above)
Comment (dzucconi) [reply src/v2/Apps/Artwork/Components/ArtworkImageBrowser/ArtworkSharePanel.tsx:134]: Yeah, `Clickable` is just a button with default styling turned off and box mixed in.

File: src/v2/Apps/Artwork/Components/TrustSignals/TrustSignal.tsx (hunk 3 above)
Comment (zephraph) [reply src/v2/Apps/Artwork/Components/TrustSignals/TrustSignal.tsx:24]: I can try!

File: src/v2/Apps/Artwork/Components/ArtworkRelatedArtists.tsx [hunk 4]
```diff
@@ -53,9 +53,9 @@ export const ArtworkRelatedArtists: React.FC<ArtworkRelatedArtistsProps> = track
     return (
       <Box mt={6} data-test={ContextModule.relatedArtistsRail}>
         <Flex flexDirection="column" alignItems="center">
-          <Serif size={["5t", "8"]} color="black100" mb={2} textAlign="center">
```
Comment (anandaroop) [inline src/v2/Apps/Artwork/Components/ArtworkRelatedArtists.tsx:56]: I did this as well (dropped the responsive props) on the assumption that `Text` will do the desired responsive thing. 

I'm assuming that's fine, but should have checked beforehand!

File: src/v2/Apps/Artwork/Components/PricingContextModal.tsx [hunk 5]
```diff
@@ -59,19 +59,19 @@ export class PricingContextModal extends React.Component<State> {
           }
         >
           <Spacer mt={2} />
-          <Serif size="3" color="black80">
+          <Text variant="text" color="black80">
```
Comment (anandaroop) [inline src/v2/Apps/Artwork/Components/PricingContextModal.tsx:62]: **suggestion, (non-blocking)**: I believe `black80` is [deprecated](https://www.notion.so/artsy/Color-a0c24896daf8433d9409aee2146ac267#c19e89075a1e45c2b8640f6d22ff48ad)? Nbd but could change here and throughout this file.

**Review by anandaroop — APPROVED** [review]
Amazing, thanks @zephraph!

One small suggestion about deprecated colors, as long as we're in here. Up to you though.

**Comment by anandaroop** [comment]
Tests updated. Merging now. Thanks for this 🎉
